src/python/gem5/simulate/simulator.py

```diff
@@ -282,12 +283,71 @@ def set_max_ticks(self, max_tick: int) -> None:
             raise ValueError(
                 f"Max ticks must be less than {m5.MaxTick}, not {max_tick}"
             )
+        if self._hypercall_max_ticks:
+            warn(
+                "A hypercall 6 exit has already been scheduled for tick "
+                f"{self._hypercall_max_ticks}. Setting hypercall and classic "
+                "exits in the same simulation is not well tested and the "
+                "simulation may not behave as expected."
+            )
+
         self._max_ticks = max_tick
 
     def get_max_ticks(self) -> int:
         assert hasattr(self, "_max_ticks"), "Max ticks not set"
         return self._max_ticks
 
+    def set_hypercall_absolute_max_ticks(
+        self, max_tick: int, exit_str: str
```

Comment:
How about 
```suggestion
        self, max_tick: int, exit_str: str="Max ticks reached"
```

And you can drop the "This must be provided to exit with hypercall 6 exit."

In the long run, I'd like to deprecate the old way of doing it, and I think having a default string will make it easier.

Maybe also `warn` if the string is empty.

Reply:
I made the suggested changes and tested. If the user passes "" for `exit_str`, the function will still trigger hypercall 6. However, if they use "Tick exit reached", which is the default string in m5.scheduleTickExitAbsolute and m5.scheduleTickExitCurrent, they will trigger `ExitEvent.SCHEDULED_TICK`. As such, I've added a warning for if users accidentally use "Tick exit reached".